src/Main.hs

```diff
@@ -90,9 +90,22 @@ getExecutables = do
         [] -> error "No executables found"
         ds -> map fst ds
 
+getCabalProjectRootCurrent :: IO (Maybe FilePath)
+getCabalProjectRootCurrent = flip catchIOError (const (return Nothing)) $ do
+  cwd <- getCurrentDirectory
+  es  <- getDirectoryContents cwd
+  if any (isSuffixOf ".cabal") es
+    then return $ Just cwd
+    else do
+      let parent = takeDirectory cwd
+      if parent == cwd
+        then return Nothing
+        else withCurrentDirectory parent getCabalProjectRootCurrent
```

Comment:
`withCurrentDictory` is the reason the build is failing; it isn't defined.

I'd rather have two functions: `getCabalProjectRootCurrent :: IO (Maybe FilePath)` and `getCabalProjectRoot :: FilePath -> IO (Maybe FilePath)`

Then the recursive call can just call the second version with `parent`, whereas the `getCabalProjectRootCurrent` version will just call the first with the current directory, without changing the environment or anything like that.